extensions/data-transfer/portability-data-transfer-imgur/src/main/java/org/datatransferproject/datatransfer/imgur/photos/ImgurPhotosExporter.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2019 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.datatransfer.imgur.photos;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.spi.transfer.provider.ExportResult;
+import org.datatransferproject.spi.transfer.provider.Exporter;
+import org.datatransferproject.types.common.ExportInformation;
+
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.datatransferproject.types.common.models.photos.PhotoAlbum;
+import org.datatransferproject.types.common.models.photos.PhotoModel;
+import org.datatransferproject.types.common.models.photos.PhotosContainerResource;
+
+import java.io.IOException;
+import java.util.*;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+/** Exports Imgur albums and photos using Imgur API */
+public class ImgurPhotosExporter
+    implements Exporter<TokensAndUrlAuthData, PhotosContainerResource> {
+  private final String albumPhotosUrlTemplate;
+  private final String allPhotosUrlTemplate;
+  private final String albumsUrlTemplate;
+  private final String defaultAlbumId;
+
+  private final OkHttpClient client;
+  private final ObjectMapper objectMapper;
+  private Monitor monitor;
+
+  public ImgurPhotosExporter(Monitor monitor, String baseUrl) {
+    albumPhotosUrlTemplate = baseUrl + "/album/%s/images";
+    albumsUrlTemplate = baseUrl + "/account/me/albums/%s";
+    allPhotosUrlTemplate = baseUrl + "/account/me/images/%s";
+    defaultAlbumId = "0";
+    this.client = new OkHttpClient.Builder().build();
```

Comment:
I think you can pass in the extension context to the constructor and get the Http Transport and object mapper from there (or pass them in directly)

 For an example see: https://github.com/google/data-transfer-project/blob/master/extensions/data-transfer/portability-data-transfer-instagram/src/main/java/org/datatransferproject/transfer/instagram/InstagramTransferExtension.java#L69

Reply:
Done, but I used `OkHttpClient` and had to register it in the [WorkerMain](https://github.com/google/data-transfer-project/blob/2dc1f8ed03e9e52ffe93bdca7c81878b5041d714/portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerMain.java#L81) 
It's used in MicrosoftPhotosExporter too. I can replace it to the `HttpTransport`, if it's preffered